src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -228,7 +228,7 @@ mod test {
         ];
 
         // for ceil decimal to int.
-        let mut ctx = EvalContext::new(Arc::new(EvalConfig::new(FLAG_IGNORE_TRUNCATE).unwrap()));
+        let mut ctx = EvalContext::new(Arc::new(*EvalConfig::new().set_truncate_as_warning(true)));
```

Comment:
CI blames `cannot move out of borrowed content`, what about changing signatures to `fn foo(mut self, bar) -> Self`?

Reply:
Then the semantic becomes copying. Are you sure?

Reply:
My suggestion is based on:

```
rror[E0507]: cannot move out of borrowed content
    --> src/coprocessor/codec/datum.rs:1694:19
     |
1694 |         let cfg = *EvalConfig::new().set_truncate_as_warning(true);
     |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |                   |
     |                   cannot move out of borrowed content
     |                   help: consider using a reference instead: `&*EvalConfig::new().set_truncate_as_warning(true)`
```

You can keep `foo(&mut self) -> &mut Self`:

```rust
let cfg = EvalContext::new();
cfg.set_a().set_b();
```

Or you can change to `foo(self) -> Self`

```rust
let cfg = EvalContext::new().set_a().set_b();
```

Also I believe compiler will optimize copying away.

Reply:
This is making TiKV depending on compiler optimization, it's not nice (like debug build might be unnecessarily slow).
I'll use a variable here.